src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -20,18 +20,22 @@ function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3React
   return [connector, hooks]
 }
 
-function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
+function getWallet(setError: (error: Error) => void, provider?: JsonRpcProvider | Eip1193Provider) {
   if (!provider) return
   if (JsonRpcProvider.isProvider(provider)) {
     return toWeb3Connection(initializeConnector((actions) => new Url({ actions, url: provider })))
   } else if (JsonRpcProvider.isProvider((provider as any).provider)) {
     throw new Error('Eip1193Bridge is experimental: pass your ethers Provider directly')
   } else {
-    return toWeb3Connection(initializeConnector((actions) => new EIP1193({ actions, provider })))
+    return toWeb3Connection(initializeConnector((actions) => new EIP1193({ actions, provider, onError: setError })))
   }
 }
 
-function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
+function getWalletConnectConnection(
```

Comment:
Can you rename this to `getWalletFromWalletConnect` to disambiguate it from `getWallet`?

Reply:
Specify the error handler after the provider, and add a return type, so the signature should be

    function getWalletFromWalletConnect(provider: JsonRpcProvider | string, setError: (error: Error) => void, useDefault: boolean): Web3Connection {
    
This aligns it with the other function signature, `getWalletFromProvider`.